docs/learn/expert-guides/deploy-multichain-contracts.md

```diff
@@ -0,0 +1,581 @@
+
+```js title="main.js"
+// Encode function call for proxy initialization
+const encodedFunctionCallForProxyInit =
+  proxyContract.interface.encodeFunctionData('setNumber', [789]);
+
+// Precompute the address for initialized ERC1167 proxy
+const precomputedInitializedProxyAddress =
+  await lsp16UniversalFactory.computeERC1167Address(
+    precomputedAddressWithInit,
+    deploymentSalt,
+    true, // --> boolean indicating whether the contract is initializable
+    encodedFunctionCallForProxyInit, // --> bytes representing the initialize data
+  );
+
+// Deploy and initialize ERC1167 proxy
+const deployAndInitProxyTx =
+  await lsp16UniversalFactory.deployERC1167ProxyAndInitialize(
+    precomputedAddressWithInit,
+    deploymentSalt,
+    encodedFunctionCallForProxyInit,
+  );
+await deployAndInitProxyTx.wait();
+
+const initializedProxyContract = new ethers.Contract(
+  precomputedInitializedProxyAddress,
+  TargetContractABI,
+  signer,
+);
+const numberInProxyAfterInit = await initializedProxyContract.number();
+console.log('The number in the initialized proxy is: ', numberInProxyAfterInit);
+```
+
+### Generating CREATE2 Salt
+
+This section is primarily intended for advanced developers who require knowledge of the specific salt used in address generation. Developers focusing on basic deployment across multiple chains may not need this detailed information and can precompute the address of the contract to be created using the provided method of the contract.
+
+The `LSP16UniversalFactory` uses a unique approach to deploying contracts with CREATE2. Instead of using the salt directly, it combines and hashes the salt with other parameters.
+
+```js title="main.js"
+// Precompute the salt for deployment
+const providedSalt = '<salt>'; // replace with your actual salt
+
+const precomputedProvidedSalt = await lsp16UniversalFactory.generateSalt(
+  providedSalt,
+  <boolean>,                // --> boolean indicating whether the contract is initializable
+  <empty or encoded call>,  // --> bytes representing the initialize data
+);
+
+console.log('The actual salt used for deployment is: ', precomputedProvidedSalt);
+```
+
+This salt can be used by off-chain tools to detect the address where the contract will be deployed along with the init code hash and the deployer address.
+
+## Step 3: Repeating on different networks
+
+To achieve the same address on a different network, make sure to:
+
+- Run the script with the same salt specified in the script.
+- Replace `RPC_URL` with the rpc of the network you want to deploy on.
+- Make sure the `signer` has enough native tokens to cover the cost of deployment on the new network.
+
+## Final code
+
+The code will run, just need to:
+
+- Replace `<private-key>` with a private key containing funds.
```

Comment:
maybe add a warning that the file should not be comited in any repo.

On other tutorial we put the pk in .env en load it from process.env and the `.env` file is not supposed to be in git.